src/Node.ts

```diff
@@ -1145,12 +1150,20 @@ export class Node {
         this.fields()[0].setValue(e.value);
     }
 
-    addEmptyField = (selectedNode:Node) :void => {
-        selectedNode.addField(new Field("New Parameter", "", "", "", "", false, Eagle.DataType.String, false, [], false));
+    addEmptyField = (index:number) :void => {
+        if(index === -1){
```

Comment:
The "new Field(...)" code is the same in both cases, so perhaps create the new field once outside the if, and then using it inside the if becomes cleaner.